app/script/event/preprocessor/QuotedMessageMiddleware.js

```diff
@@ -75,13 +75,16 @@ z.event.preprocessor.QuotedMessageMiddleware = class QuotedMessageMiddleware {
 
   _handleEditEvent(event) {
     const originalMessageId = event.data.replacing_message_id;
-    return this._findRepliesToMessage(event.conversation, originalMessageId).then(replies => {
+    return this._findRepliesToMessage(event.conversation, originalMessageId).then(({originalEvent, replies}) => {
       this.logger.info(`Updating '${replies.length}' replies to updated message '${originalMessageId}'`);
+
       replies.forEach(reply => {
         reply.data.quote.message_id = event.id;
         this.eventService.replaceEvent(reply);
       });
-      return event;
+
+      const decoratedData = Object.assign(event.data, {quote: originalEvent.data.quote});
+      return Object.assign(event, {data: decoratedData});
```

Comment:
You are assigning the `{quote: ...}` to `event.data` already. So I don't think it's necessary to do `Object.assign(event, {data: decoratedData})`.